# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/svg/SVGUtils.h
# Commit: ec44194caefb
# Full Hash: ec44194caefb41931247d18ad886326cf6d89a93
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2024-03-05 09:48:50
# Description:
#   Bug 1870200 - Break the cyclic dependency if any of the desendants uses non-scaling-stroke. r=emilio
#   
#   While we are computing the transform-box:stroke-box, for CSS Transforms or
#   Motion path Transforms, we have to avoid the cyclic dependency not only
#   for the SVG geometry frame itself, but also for the desendants of
# ==============================================================================

diff -r 5c84d062c789 -r ec44194caefb layout/svg/SVGUtils.h
--- a/layout/svg/SVGUtils.h	Mon Mar 04 20:33:13 2024 +0000
+++ b/layout/svg/SVGUtils.h	Mon Mar 04 20:46:05 2024 +0000
@@ -346,6 +346,12 @@
     // For a frame with a clip-path, if this flag is set then the result
     // will not be clipped to the bbox of the content inside the clip-path.
     eDoNotClipToBBoxOfContentInsideClipPath = 1 << 10,
+    // For some cases, e.g. when using transform-box: stroke-box, we may have
+    // the cyclical dependency if any of the elements in the subtree has
+    // non-scaling-stroke. In this case, we should break it and use
+    // transform-box:fill-box instead.
+    // https://github.com/w3c/csswg-drafts/issues/9640
+    eAvoidCycleIfNonScalingStroke = 1 << 11,
   };
   /**
    * This function in primarily for implementing the SVG DOM function getBBox()